tests/pthread/call_async_on_main_thread.js

```diff
@@ -3,7 +3,7 @@ mergeInto(LibraryManager.library, {
   report_result__proxy: 'async',
   report_result__sig: 'viii',
   report_result: function(param1, param2, param3) {
-    if (ENVIRONMENT_IS_WORKER) {
+    if (typeof ENVIRONMENT_IS_WORKER !== 'undefined' && ENVIRONMENT_IS_WORKER) {
```

Comment:
This test change and all the test changes below are to make pthreads tests pass in MINIMAL_RUNTIME. It enables a workflow where I locally edit `src/settings.js` directly to change `var MINIMAL_RUNTIME = 1;` by default to run the whole browser.test_pthread* test suite in MINIMAL_RUNTIME, even though we don't duplicate all browser tests twice for MINIMAL_RUNTIME and non-MINIMAL_RUNTIME cases (only select tests are duplicated)